presto-main/src/main/java/io/prestosql/operator/exchange/PageReference.java

```diff
@@ -17,24 +17,28 @@
 
 import javax.annotation.concurrent.ThreadSafe;
 
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.function.LongConsumer;
 
+import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
-class PageReference
+final class PageReference
```

Comment:
unrelated change. Extract to separate commit

Reply:
It's material to the change to using `AtomicIntegerFieldUpdater` which will not work as expected with subclasses.